docs/pages/versions/unversioned/workflow/testing-with-jest.md

```diff
@@ -0,0 +1,171 @@
+
+```js
+"scripts": {
+  ...
+  "test": "jest"
+},
+"jest": {
+  "preset": "jest-expo"
+}
+```
+
+Now let's add react-test-renderer to our project:
+
+`yarn add react-test-renderer --dev` **or** `npm i react-test-renderer --save-dev`
+
+That's it! Now we can start writing Jest tests!
+
+> **Note**: [react-native-testing-library](https://github.com/callstack/react-native-testing-library) is a library build on top of react-test-renderer that could be helpful in your workflow.
+
+## Unit Test
+
+We are going to write a simple test for `App.js` by creating `App.test.js`. Jest will identify this as a test, and include it in the tests queue. There are other ways to [structure your tests](#structure-your-tests), but we will cover that later in this tutorial.
+
+Our test will be the expected state of the `<App />` to have 1 child element:
+
+```js
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import App from './App';
+
+describe('<App />', () => {
+  it('has 1 child', () => {
+    const tree = renderer.create(<App />).toJSON();
+    expect(tree.children.length).toBe(1);
+  });
+});
+```
+
+Now run `yarn test` or `npm run test`, if all went well you should see 1 test passed! Read more on [expect and conditional matchers](https://jestjs.io/docs/en/expect).
+
+> **Node Version**: If you are running node version 11.11, [Jest will throw errors](https://github.com/facebook/jest/issues/8069), please downgrade or upgrade to 11.12+
+
+## Snapshot Test
+
+Now let's add a snapshot test for `App.js`. **What is a snapshot test, and why is it useful?** Snapshot tests are used to make sure the UI stays consistent, especially when a project is working with global styles that are potentially shared across components. Read more about it on Jest's site [snapshot testing](https://jestjs.io/docs/en/snapshot-testing).
+
+Let's add the following within the describe():
+```js
+it('renders correctly', () => {
+  const tree = renderer.create(<App />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+```
+
+Now run `yarn test` or `npm run test`, if all went well you should see a snapshot created and 2 tests passed!
+
+This was a very simple test, for more information take a look at the following links:
+
+- [Api: Globals](https://jestjs.io/docs/en/api)
+- [Api: Mock Functions](https://jestjs.io/docs/en/mock-function-api)
+- [Api: Expect](https://jestjs.io/docs/en/expect)
+- [Cli Options](https://jestjs.io/docs/en/cli)
+
+## Code Coverage Reports
+
+Running tests are cool and all, but wouldn't you like to see your Expo projects total code coverage?! Maybe even see it in html format?! This section is for you!
+
+Let's head back to `package.json` and add the following:
+
+```js
+"jest": {
+  ...
+  "collectCoverage": true,
+  "collectCoverageFrom": [
+    "**/*.{js,jsx}",
+    "!**/coverage/**",
+    "!**/node_modules/**",
+    "!**/babel.config.js",
+    "!**/jest.setup.js"
+  ],
+  "verbose": true
```

Comment:
why add `verbose`?

Reply:
i guess i was thinking testDebug would be the only one that verbose would be set to false. should it not be true by default?